src/screens/ReaderScreen.tsx

```diff
@@ -1,13 +1,27 @@
 import * as React from "react";
 import { FunctionComponent } from 'react';
-import { SafeAreaView, StyleSheet } from 'react-native';
+import { Dimensions, SafeAreaView, StyleSheet } from 'react-native';
 import { ReaderScreenProps } from '../navigationTypes';
 import BookReader from "../BookReader";
 
+const windowDimensions = Dimensions.get('window');
+
 export const ReaderScreen: FunctionComponent<ReaderScreenProps> = ({ route, navigation }: ReaderScreenProps) => {
+    const [windowSize, setWindowSize] = React.useState(windowDimensions);
+
+    React.useEffect(() => {
+        const subscription = Dimensions.addEventListener(
+            "change",
+            ({ window, screen }) => {
+                setWindowSize(window);
+            }
+        );
+        return () => subscription?.remove();
+    });
+
     return (
         <SafeAreaView style={styles.container}>
-            <BookReader bookUrl={route.params.bookUrl} />
+            <BookReader bookUrl={route.params.bookUrl} windowSize={windowSize} />
```

Comment:
I feel like it's weird that we pass these down through props.  Here are two ideas that I like better:

1) (i'm not sure if this could be pulled off).  I think it'd be preferable if we set the width/height of the SafeAreaView, and then BookReader just fills the entire width of its parent. (either... implicity, or with flex:1, or with width/height 100%, although I realize height 100% was being pretty flaky for us...

2) If that doesn't work, I don't think BookReader wants the window size per se, but rather the size the parent component wants to allocate to it. Which, in this case, we're approximating with windowSize. I guess it'd just be a matter of renaming the prop from windowSize to like... "size" or "bloomPlayerSize" or "readerSize" or "renderSize" or something like that

Reply:
Hmm... it seems like now we are using it in even more places than before? :(

Reply:
Drat! Fixed. This is what I MEANT to do! :D